catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/filter/TransformVisitor.java

```diff
@@ -77,6 +78,35 @@ public void visitLiteralType(VisitableElement<List<Serializable>> visitable) {
     builder.setValue(values.get(0).toString());
   }
 
+  @Override
+  public void visitDistanceType(VisitableElement<Double> visitable) {
+    Double value = visitable.getValue();
+    if (value == null) {
+      LOGGER.debug("No values found on distance type");
+      return;
+    }
+
+    builder.setDistance(value);
+  }
+
+  @Override
+  public void visitDistanceBufferType(VisitableElement<List<Object>> visitable) {
+    traceName(visitable);
+    List<Object> values = visitable.getValue();
```

Comment:
✏️ Depending on the implementation of whatever implements VisitableElement this could return null as is the case with `VisitableJsonElementImpl`.

Reply:
Right, so I added a `CollectionUtils.isEmpty()` check